### PR TITLE
Import latest config changes from MRCPP

### PR DESCRIPTION
**Pull request by stigrj**
I noticed that this setup fails if I use MRCPP-1.1.0, and I had to upgrade to MRCPP-master for the `MPI` and `OpenMP` configuration to work.

New output header will look like this
```
**********************************************************************
***                                                                ***
***                                                                ***
***        __  __ ____   ____ _                                    ***
***       |  \/  |  _ \ / ___| |__   ___ _ __ ___                  ***
***       | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                 ***
***       | |  | |  _ <| |___| | | |  __/ | | | | |                ***
***       |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                ***
***                                                                ***
***       VERSION            0.3.0-alpha                           ***
***                                                                ***
***       Git branch         misc-conf                             ***
***       Git commit hash    15329032ad6f4514b268                  ***
***       Git commit author  Stig Rune Jensen                      ***
***       Git commit date    Thu Jan 30 14:42:30 2020 +0100        ***
***                                                                ***
***       Magnar Bjorgve     <magnar.bjorgve@uit.no>               ***
***       Roberto Di Remigio <roberto.di.remigio@uit.no>           ***
***       Luca Frediani      <luca.frediani@uit.no>                ***
***       Gabriel Gerez      <gsa017@post.uit.no>                  ***
***       Stig Rune Jensen   <stig.r.jensen@uit.no>                ***
***       Peter Wind         <peter.wind@uit.no>                   ***
***                                                                ***
**********************************************************************

----------------------------------------------------------------------

 MPI processes         :      (no bank)                             1
 OpenMP threads        :                                            4
 Total cores           :                                            4
                                                                      
----------------------------------------------------------------------

XCFun DFT library Copyright 2009-2011 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

----------------------------------------------------------------------

 MRCPP version         : 1.2.0-alpha
 Git branch            : master
 Git commit hash       : d05e569bf9c9432d1ffa
 Git commit author     : Stig Rune Jensen
 Git commit date       : Fri Jan 24 13:14:40 2020 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP

----------------------------------------------------------------------
```

### CODE REVIEW

File: external/upstream/fetch_mrcpp.cmake [hunk 1]
```diff
@@ -11,7 +11,7 @@ else()
     GIT_REPOSITORY
       https://github.com/MRChemSoft/mrcpp.git
     GIT_TAG
-      v1.1.0
+      29cf9baa31f26b7332fb94b3d27251d3c4aa85dd
```
Comment (robertodr) [inline external/upstream/fetch_mrcpp.cmake]: How about tagging a release of MRCPP? Even just a prerelease (beta or release candidate) I think we're close enough with the packaging.

File: config.h.in [hunk 2]
```diff
@@ -23,18 +23,18 @@
  * <https://mrchem.readthedocs.io/>
  */
 
-#define PROGRAM_VERSION "@PROGRAM_VERSION@"
+#pragma once
 
-#define GIT_REVISION "@GIT_REVISION@"
+namespace mrchem {
+inline constexpr auto sad_basis_dir() noexcept {
```
Comment (robertodr) [inline config.h.in]: I suspect this needs the same duplicitous treatment (build vs install tree) `MW_FILTER_DIR` got in MRCPP. If you judge it not relevant for this PR, would you open an issue?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -5,6 +5,8 @@ add_library(mrchem
   mrenv.cpp
   )
```
Comment (robertodr) [inline src/CMakeLists.txt]: You might be missing something like:
```
target_link_libraries(mrchem
   PUBLIC 
       $<$<TARGET_EXISTS:OpenMP::OpenMP_CXX>:OpenMP::OpenMP_CXX>
       $<$<TARGET_EXISTS:MPI::MPI_CXX>:MPI::MPI_CXX>
  )
```
to consume the OpenMP and MPI targets.

File: cmake/custom/omp.cmake [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# Variables used::
+#
+#   ENABLE_OPENMP
+#
+# autocmake.yml configuration::
+#
+#   docopt: "--omp Enable OpenMP parallelization [default: False]."
+#   define: "'-DENABLE_OPENMP={0}'.format(arguments['--omp'])"
+
+option(ENABLE_OPENMP "Enable OpenMP parallelization" OFF)
```
Comment (robertodr) [inline cmake/custom/omp.cmake]: Has `HAVE_OPENMP` been replaced everywhere by `_OPENMP`?

Comment (stigrj) [reply cmake/custom/omp.cmake]: It was actually never used in mrchem, only mrcpp

Comment (robertodr) [reply cmake/custom/omp.cmake]: OK, cool.

File: src/CMakeLists.txt (hunk 3 above)
Comment (stigrj) [reply src/CMakeLists.txt]: This does not help when using MRCPP-1.1.0, and I get the same horrible `Child aborted`error in the mrchem unit tests as we saw with the conda build.

Comment (robertodr) [reply src/CMakeLists.txt]: Can you try serial, OpenMP, and hybrid? I feel like serial should work and that the problem comes from linking twice against OpenMP. I have to figure out how to export properly the flags used to build MRCPP to client projects.

File: config.h.in (hunk 2 above)
Comment (stigrj) [reply config.h.in]: Done

File: src/CMakeLists.txt (hunk 3 above)
Comment (robertodr) [reply src/CMakeLists.txt]: If you remove those and build serial, OpenMP, and hybrid, does it work correctly?

Comment (stigrj) [reply src/CMakeLists.txt]: Without `...TARGET_EXISTS...`:
MRCPP-1.1.0: serial works, but omp and hybrid fail.
MRCPP-master: everything works

Comment (robertodr) [reply src/CMakeLists.txt]: I guess detection and usage of OpenMP/MPI in MRCPP 1.1.0 is incompatible with the the one using the target. This is not entirely surprising, but then I would tag both MRChem and MRCPP and make explicit the fact that there's an incompatibility between certain versions. Maybe a search in the CMake issue board reveals something?

File: src/initial_guess/sad.cpp [hunk 5]
```diff
@@ -213,9 +213,12 @@ void initial_guess::sad::project_atomic_densities(double prec, Density &rho_tot,
 
     auto crop_prec = (mpi::numerically_exact) ? -1.0 : prec;
     std::string sad_path;
+    println(0, "source dir: " << sad_basis_source_dir());
+    println(0, "install dir: " << sad_basis_install_dir());
     for (auto n : {sad_basis_source_dir(), sad_basis_install_dir()}) {
         if (mrcpp::details::directory_exists(n)) sad_path = n;
```
Comment (robertodr) [inline src/initial_guess/sad.cpp]: You probably need a `break;` so that you use the first path that exists?

Comment (stigrj) [reply src/initial_guess/sad.cpp]: ty

File: src/CMakeLists.txt (hunk 3 above)
Comment (stigrj) [reply src/CMakeLists.txt]: Should I include the `TARGET_EXISTS` or not?

Comment (robertodr) [reply src/CMakeLists.txt]: I would say yes, but I am not sure if that's always going to be OK. I need to check it locally, but will only be able to do it in a day or two...

**Comment by stigrj** [comment]
I have now imported all the versioning stuff from MRCPP.

This should be ready now, save for the MRCPP version which currently points to latest `master`. I guess we *can* in principle point to a commit on MRCPP `master` now since we are in a alpha version of MRChem, but we should probably put together a new MRCPP release.

**Comment by ilfreddy** [comment]
If you think it is time for a new release, go ahead.

**Comment by robertodr** [comment]
> I have now imported all the versioning stuff from MRCPP.
> 
> This should be ready now, save for the MRCPP version which currently points to latest `master`. I guess we _can_ in principle point to a commit on MRCPP `master` now since we are in a alpha version of MRChem, but we should probably put together a new MRCPP release.

I vote for a new release

**Comment by stigrj** [comment]
Now updated to `MRCPP-v1.2.0-alpha1`

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -84,6 +84,8 @@ target_link_libraries(mrchem.x
     MRCPP::mrcpp
     XCFun::xcfun
     nlohmann_json::nlohmann_json
+    $<$<TARGET_EXISTS:MPI::MPI_CXX>:MPI::MPI_CXX>
+    $<$<TARGET_EXISTS:OpenMP::OpenMP_CXX>:OpenMP::OpenMP_CXX>
```
Comment (robertodr) [inline src/CMakeLists.txt]: Given that this bunch is `PUBLIC` for `mrchem`, it should transitively be linked to `mrchem.x` automatically.

**Pull request merged**